src/internal/obj/minio_client.go

```diff
@@ -46,11 +47,19 @@ func newMinioClientV2(endpoint, bucket, id, secret string, secure bool) (*minioC
 
 func (c *minioClient) Put(ctx context.Context, name string, r io.Reader) (retErr error) {
 	defer func() { retErr = c.transformError(retErr, name) }()
-	wc := newMinioWriter(ctx, c, name)
-	if _, err := io.Copy(wc, r); err != nil {
+	opts := minio.PutObjectOptions{
+		ContentType: "application/octet-stream",
+		PartSize:    uint64(8 * 1024 * 1024),
+	}
+	n, err := c.Client.PutObjectWithContext(ctx, c.bucket, name, r, -1, opts)
+	if err != nil {
+		return err
+	}
+	if n == 0 {
```

Comment:
what's going on here?  could probably use a comment

Reply:
I was trying to fix the minio client to support 0 length objects.  This didn't fix it, so I will just remove it.